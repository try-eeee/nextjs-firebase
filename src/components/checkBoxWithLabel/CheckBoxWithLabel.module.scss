.checkbox {
  margin: 16px 0 8px 0;
  font-size: 14px;
  display: flex;
  align-items: center;

  input {
    margin-right: 8px;
  }

  .label {
    a {
      color: #007bff;
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }
}
