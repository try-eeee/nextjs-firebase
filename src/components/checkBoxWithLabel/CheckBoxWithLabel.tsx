import React from "react";
import { UseFormRegisterReturn } from "react-hook-form";
import styles from "./CheckBoxWithLabel.module.scss";

interface CheckBoxWithLabelProps
  extends React.ComponentPropsWithoutRef<"input"> {
  children: React.ReactNode;
  register?: UseFormRegisterReturn;
}

export function CheckBoxWithLabel(props: CheckBoxWithLabelProps) {
  const { children, register, ...rest } = props;
  return (
    <label className={styles.checkbox}>
      <input type="checkbox" {...register} {...rest} />
      <span className="label">{children}</span>
    </label>
  );
}
