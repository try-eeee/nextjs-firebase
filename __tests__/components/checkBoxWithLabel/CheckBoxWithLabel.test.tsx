import React from "react";
import { render } from "@testing-library/react";
import { CheckBoxWithLabel } from "@/components/checkBoxWithLabel/CheckBoxWithLabel";

describe("CheckBoxWithLabel UnitTest", () => {
  test("Snapshot Test Of CheckBoxWithLabel", () => {
    const { asFragment } = render(
      <CheckBoxWithLabel>
        <>
          利用規約に同意する 利用規約は
          <a href="#" target="_blank" rel="noopener noreferrer">
            こちら
          </a>
        </>
      </CheckBoxWithLabel>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
