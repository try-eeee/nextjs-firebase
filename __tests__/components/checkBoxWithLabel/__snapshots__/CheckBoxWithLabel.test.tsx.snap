// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxWithLabel UnitTest Snapshot Test Of CheckBoxWithLabel 1`] = `
<DocumentFragment>
  <label
    class="checkbox"
  >
    <input
      type="checkbox"
    />
    <span
      class="label"
    >
      利用規約に同意する 利用規約は
      <a
        href="#"
        rel="noopener noreferrer"
        target="_blank"
      >
        こちら
      </a>
    </span>
  </label>
</DocumentFragment>
`;
